### ISSUE TITLE
Incorrect coverage

### ISSUE THREAD

**Issue by ofek** [codecov/support#448]
@stevepeak @304NotModified @nfischer @skirpichev @mpavlov @rockuw Issue #191 is happening to me too and I think I've discovered the reason why this is happening to people only sometimes.

Recent commits:
![rc](https://user-images.githubusercontent.com/9677399/30436245-a6f4a102-9939-11e7-952e-fd906c594718.PNG)

That first one isn't even committed; it's just a PR.

Now observe coverage:
![cb](https://user-images.githubusercontent.com/9677399/30436106-63778782-9939-11e7-8dd4-c1a25b54a4d0.PNG)

My most recent commit to master is 100%:
![mc](https://user-images.githubusercontent.com/9677399/30436404-075007c6-993a-11e7-8efd-6629ff89b967.PNG)

Though the badge shows the PR:
![pr](https://user-images.githubusercontent.com/9677399/30436657-afc452ea-993a-11e7-98b4-98867563d10c.PNG)

Therefore, from what I can tell, Codecov reports incorrect stats when a PR is opened directly from a contributor's master instead of from a branch. It can't differentiate between the 2 master branches.

**Comment by skirpichev** [comment]
> It can't differentiate between the 2 master branches.

Maybe, but for sure - my PR's, mentioned in #191 were not on the master branch.

**Comment by nfischer** [comment]
I saw incorrect coverage when opening a PR from a feature branch (on the main repo). The PR was landing on master.

**Comment by JimBobSquarePants** [comment]
Wildly inaccurate. I've got a 6.43% decrease with zero code changes. 

https://github.com/SixLabors/ImageSharp/pull/363/files

**Comment by johlju** [comment]
@ofek I think I hit the same issue? It seems that a PR that is sent in, which for example does not have proper testing and percentage goes down, because the badge is (wrongly) calculating (getting percent coverage) on the PR instead of the branch.

Latest merge/commit had coverage at 100%.

![image](https://user-images.githubusercontent.com/7189721/32088008-4e8af214-bae0-11e7-95a3-5770797ff9a7.png)

Badge is today showing 98% coverage.

![image](https://user-images.githubusercontent.com/7189721/32088019-67bdcedc-bae0-11e7-88b6-b7bb44887b71.png)

Pull request https://github.com/PowerShell/xFailOverCluster/pull/153 was sent, without tests, so that has 2 percent less coverage.

![image](https://user-images.githubusercontent.com/7189721/32087973-16ce323c-bae0-11e7-89f7-e68e48ce0bad.png)

**Comment by johlju** [comment]
@ofek and the contributor is using the dev branch as the working branch  - so I'm seeing the same behavior.

**Comment by ofek** [comment]
Yes and look same branch name `dev`. I think that is the issue (or perhaps one of multiple).

**Comment by faheel** [comment]
I can confirm this bug: if there's a PR from a fork's X branch against upstream X branch, the website and the badge and both report the latest coverage to be the coverage of that PR.

**Comment by stevepeak** [comment]
Hey fellows. I want to check on this issue to see if there are any new pull requests I can review. Thank you for your patience.

**Comment by stevepeak** [comment]
@faheel do you have an example commit I can review to see this in action?

**Comment by faheel** [comment]
@stevepeak, I received [this PR](https://github.com/faheel/BigInt/pull/14) (with a single commit) on a project of mine when I found out this problem. The commit in the PR was made on the fork's `master` branch. I noticed that the coverage badge that I have in the README was showing a lower coverage % which was from the PR, and when I checked out the [project page on Codecov](https://codecov.io/gh/faheel/BigInt), it too reported the same coverage from the PR in the files section at the bottom (on the main project page and not on the commit's page).

After that I pushed a few commits myself on the `master` branch and updated coverage was correctly reported.

If you want to test it yourself, go ahead. Fork [this repo](https://github.com/faheel/BigInt), delete one or more `TEST_CASE()` from one or more `*.test.cpp` files in the `test` directory, commit to `master`, send a PR, and once the Travis build completes (takes about 15 mins) check out the Codecov page and the badge in the README.
